monkey/agent_plugins/payloads/ransomware/src/readme_dropper.py

```diff
@@ -2,13 +2,35 @@
 import shutil
 from pathlib import Path
 
+from common import OperatingSystem
+
 logger = logging.getLogger(__name__)
 
 
-def leave_readme(src: Path, dest: Path):
-    if dest.exists():
-        logger.warning(f"{dest} already exists, not leaving a new README.txt")
-        return
+class ReadmeDropper:
+    def __init__(self, operating_system: OperatingSystem):
+        self._operating_system = operating_system
+
+    def leave_readme(self, src: Path, dest: Path):
+        if dest.exists():
+            logger.warning(f"{dest} already exists, not leaving a new README.txt")
+            return
+
+        logger.info(f"Leaving a ransomware README file at {dest}")
+
+        if self._operating_system == OperatingSystem.WINDOWS:
+            self._leave_windows_readme(src, dest)
+        else:
+            self._leave_posix_readme(src, dest)
```

Comment:
Could just do
```python
class ReadmeDropper:
    def __init__(self, operating_system: OperatingSystem):
        if operating_system == OperatingSystem.WINDOWS:
            self._leave_readme = self._leave_windows_readme
        else:
            self._leave_readme = self._leave_posix_readme

    def leave_readme(self, src: Path, dest: Path):
        if dest.exists():
            logger.warning(f"{dest} already exists, not leaving a new README.txt")
            return

        logger.info(f"Leaving a ransomware README file at {dest}")
        
        self._leave_readme(src, dest)
```

Reply:
As written, it's less cognitive load on the developer to be able to read `leave_readme()` and understand exactly what's happening.